### PR TITLE
Update lib to use prices 1.0.0

### PR DESCRIPTION
**Pull request by rafalp**
This PR bumps lib's version to 1.0.0, redoes the api to be function based with functions returning lists of payloads to send to Google's API, and updates codebase to work with `TaxedMoney`.

**TODO**

- [x] Convert `EnhancedItem` and `EnhancedTransaction` to functions
- [x] Add `**extra_data` to all functions
- [x] Split `__init__` into few smaller packages
- [x] Move tests suite to `py.test`
- [x] Update readme

### CODE REVIEW

**Comment by patrys** [comment]
Can we please have all items accept `**kwargs` and merge them into the the resulting dict? Adding custom fields is a common case and is currently fairly awkward.

File: google_measurement_protocol/__init__.py [hunk 1]
```diff
@@ -1,256 +1,9 @@
...
-        yield to_return
+from .enhanced_purchase import enhanced_item, enhanced_purchase
+from .event import event
+from .pageview import pageview
+from .report import report
+from .transaction import item, transaction
+
+__all__ = [
+    'enhanced_item', 'enhanced_purchase', 'event', 'pageview', 'report',
+    'item', 'transaction']
```
Comment (patrys) [inline google_measurement_protocol/__init__.py]: Nitpick: newline missing.

File: google_measurement_protocol/enhanced_purchase.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    yield payload
+
+
+def _finalize_enhanced_purchase_item(item, position):
+    position_prefix = 'pr{0}'.format(position)
+    final_item = {}
+    for key, value in item.items():
+        final_item[position_prefix + key] = value
+    return final_item
```
Comment (patrys) [inline google_measurement_protocol/enhanced_purchase.py]: Nitpick: newline missing.

File: google_measurement_protocol/event.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+def event(category, action, label=None, value=None, **extra_data):
+    payload = {'t': 'event', 'ec': category, 'ea': action}
+    if label:
+        payload['el'] = label
+    if value:
+        payload['ev'] = str(int(value))
+    payload.update(extra_data)
+    yield payload
```
Comment (patrys) [inline google_measurement_protocol/event.py]: Nitpick: newline missing.

File: google_measurement_protocol/pageview.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        payload['dp'] = path
+    if title:
+        payload['dt'] = title
+    if referrer:
+        payload['dr'] = referrer
+    if language:
+        payload['ul'] = language
+
+    payload.update(extra_data)
+    yield payload
```
Comment (patrys) [inline google_measurement_protocol/pageview.py]: Nitpick: newline missing.

File: google_measurement_protocol/report.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    Updates payloads setting required non-specific values on data.
+    """
+    extra_payload = {
+        'v': '1', 'tid': tracking_id, 'cid': client_id, 'aip': '1'}
+
+    for payload in payloads:
+        final_payload = dict(payload)
+        final_payload.update(extra_payload)
+        final_payload.update(extra_data)
+        yield final_payload
```
Comment (patrys) [inline google_measurement_protocol/report.py]: Nitpick: newline missing.

File: google_measurement_protocol/transaction.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    if tax is not None:
+        payload['tt'] = str(tax.amount)
+        
+    payload.update(extra_data)
+    yield payload
+    
+    for item in items:
+        final_item = dict(item)
+        final_item['ti'] = transaction_id
+        yield final_item
```
Comment (patrys) [inline google_measurement_protocol/transaction.py]: Nitpick: newline missing.

File: tests/test_enhanced_purchase.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+def test_extra_params():
+    items = [enhanced_item('item-01', Money(10, 'USD'))]
+    generator = enhanced_purchase(
+        'trans-01', items, Money(10, 'USD'), '/cart/', ex='extra')
+    assert list(generator) == [
+        {'t': 'event', 'ec': 'ecommerce', 'ea': 'purchase'},
+        {
+            'pa': 'purchase', 'ti': 'trans-01', 'dp': '/cart/', 'tt': '0',
+            'tr': '10', 'ex': 'extra', 'pr1nm': 'item-01', 'pr1pr': '10',
+            'pr1qt': 1}]
```
Comment (patrys) [inline tests/test_enhanced_purchase.py]: Nitpick: newline missing.

File: tests/test_event.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    assert list(generator) == [
+        {
+            't': 'event', 'ec': 'category', 'ea': 'action', 'el': 'label',
+            'ev': '7'}]
+
+
+def test_extra_params():
+    generator = event('category', 'action', ex='extra')
+    assert list(generator) == [
+        {'t': 'event', 'ec': 'category', 'ea': 'action', 'ex': 'extra'}]
```
Comment (patrys) [inline tests/test_event.py]: Nitpick: newline missing.

File: tests/test_item.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    data = item('item-01', Money(10, 'USD'), item_id='it01', category='cat')
+    assert data == {
+        't': 'item', 'in': 'item-01', 'ip': '10', 'cu': 'USD', 'ic': 'it01',
+        'iv': 'cat'}
+
+
+def test_extra_params():
+    data = item('item-01', Money(10, 'USD'), ex='extra')
+    assert data == {
+        't': 'item', 'in': 'item-01', 'ip': '10', 'cu': 'USD', 'ex': 'extra'}
```
Comment (patrys) [inline tests/test_item.py]: Nitpick: newline missing.

File: tests/test_pageview.py [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+def test_optional_params():
+    generator = pageview('/', title='title', referrer='referrer')
+    assert list(generator) == [
+        {'t': 'pageview', 'dp': '/', 'dr': 'referrer', 'dt': 'title'}]
+
+
+def test_extra_params():
+    generator = pageview('/my-page/', ex='extra')
+    assert list(generator) == [
+        {'t': 'pageview', 'dp': '/my-page/', 'ex': 'extra'}]
```
Comment (patrys) [inline tests/test_pageview.py]: Nitpick: newline missing.

File: tests/test_report.py [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    assert data == {
+        'cid': ['CID'], 'tid': ['UA-123456-78'], 'v': ['1'], 'aip': ['1'],
+        't': ['mock']}
+
+
+@with_httmock(ga_mock)
+def test_extra_params():
+    (response,) = report('UA-123456-78', 'CID', [{'t': 'mock'}], ex='extra')
+    data = response.json()
+    assert data['ex'] == ['extra']
```
Comment (patrys) [inline tests/test_report.py]: Nitpick: newline missing.

File: tests/test_transaction.py [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+def test_extra_params(single_item_list):
+    generator = transaction(
+        'trans-01', single_item_list, Money(10, 'USD'), ex='extra')
+    assert list(generator) == [
+        {
+            't': 'transaction', 'ti': 'trans-01', 'tr': '10', 'tt': '0',
+            'cu': 'USD', 'ex': 'extra'},
+        {
+            't': 'item', 'in': 'item-01', 'ip': '10', 'cu': 'USD',
+            'ti': 'trans-01'}]
```
Comment (patrys) [inline tests/test_transaction.py]: Nitpick: newline missing.

**Comment by rafalp** [comment]
@patrys 

> Question: would it make sense add type information now that we're dropping Python 2?

Most of API is returning dicts, but typing functions params will be big usability win here and not that hard to do (just copy param types from google docs), so I'll do another pass with those.

**Pull request merged**